sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -271,6 +272,11 @@ public long putLog(@NonNull Log log, @NonNull String group, @IntRange(from = Fla
                 targetKey = null;
                 targetToken = null;
             }
+            long maxSize = mDatabaseManager.getMaxSize();
+            if (!isLargePayload && maxSize < payloadSize) {
```

Comment:
We definitely know maxSize == payloadSize will drop all logs due to other information stored with the log. `<=` in the condition?

Reply:
The change is made [here](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1222) for Apple SDK.